File: State/cpp-source/states/Red.cpp
//
// Created by Pavel Akhtyamov on 2019-04-23.
//

#include "Red.h"
#include "Yellow.h"
#include "../TrafficLight.h"

void Red::Switch(TrafficLight *light) {
  light->SetState(std::make_unique<Yellow>());
  light->SetDirection(Direction::TO_GREEN);
}

std::string Red::GetStringState() {
  return "Red";
}

File: JavaSource/src/main/java/ru/akhcheck/patterns/state/states/Red.java
package ru.akhcheck.patterns.state.states;

import ru.akhcheck.patterns.state.Direction;
import ru.akhcheck.patterns.state.TrafficLight;

/** Красный свет: следующим всегда идёт жёлтый в сторону зелёного. */
public class Red implements State {
    @Override
    public void switchState(TrafficLight light) {
        light.setState(new Yellow());
        light.setDirection(Direction.TO_GREEN);
    }

    @Override
    public String getStringState() {
        return "Red";
    }
}
